### DISCUSSION TITLE
How to use with MaterialExtension?

### DISCUSSION THREAD

**Discussion by mekroner** [jakobhellermann/bevy-inspector-egui#191] [Q&A]
I want to change the values (`depth`) of my Extended Material using the inspector. 
This is the material: 
```rust
type WaterMaterial = ExtendedMaterial<StandardMaterial, WaterExtension>;

#[derive(Asset, AsBindGroup, Reflect, Debug, Clone)]
#[reflect(Asset)]
struct WaterExtension {
    #[uniform(50)]
    depth: f32,
}
```
Then I registered the Extension:
``` rust
fn build(&self, app: &mut App) {
    app.register_type::<WaterExtension>()
```
In the inspector I then find `WaterExtension`, but I get the message " TypeId has no ReflectHandle Type data, so it cannot be displayed" and can't see the data.
Registering the `WaterMaterial` did nothing.

How do I fix this?

**Comment by SK83RJOSH** [comment]
I can't get it to work with a custom types at all. Seems any kind of `Handle<CustomType>` simply doesn't work, which is unfortunately very limiting and frustrating. Looking at the source, it seems like `InspectorPrimitive` is implemented manually for a common `Handle` types - which you can't do as an end user since `Handle` and `InspectorPrimitive` are both external...

Nevermind the fact though, it feels like this should just work? It seems to work fine with custom any `Resource`, but not a custom `Asset` and its `Handle` - which makes me think that perhaps this side of the plugin is unfinished/needing work as I can't find any information on the topic whatsoever.
